belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/api/InputValidationIssue.java

```diff
@@ -288,18 +307,42 @@ public InputValidationIssue valueEntry(String key, Object value) {
         return this;
     }
 
-    public InputValidationIssue input(Input<?> input) {
-        addInput(input);
-        return this;
-    }
-
+    /**
+     *
+     * @param inputs list with input items to initialize the inputs parameter. Any previous inputs are erased from the
+     *        parameter
+     * @return this
+     *
+     * @throws IllegalArgumentException if in/name/value parameters are not null (because it's mutually exclusive). This
+     *         exception is also thrown if the list contains only one non-null item.
+     */
     public InputValidationIssue inputs(List<Input<?>> inputs) {
-        setInputs(inputs);
+
+        if (inputs == null) {
+            return this;
+        }
+
+        if (hasInNameValue()) {
+            throw new IllegalArgumentException(INPUTS_AND_IN_NAME_VALUE_ARE_MUTUALLY_EXCLUSIVE);
+        }
+
+        List<Input<?>> filteredInputs = inputs.stream().filter(Objects::nonNull).collect(Collectors.toList());
+
+        if (filteredInputs.size() == 1) {
+            throw new IllegalArgumentException(INPUTS_SETTER_ONE_ITEM);
+        }
+
+        this.inputs.clear();
+        this.inputs.addAll(filteredInputs);
         return this;
     }
 
     public InputValidationIssue inputs(Input<?>... inputs) {
```

Comment:
As we want to enforce at least two inputs, maybe we could change the method signature to

`inputs(Input<?> first, Input<?> second, Input<?>... otherInputs)`

which will catch the error at compile time rather than runtime?

Do you see any drawbacks with that @usku01 @clone1612?

Reply:
> Do you see any drawbacks with that @usku01 @clone1612?

I'm ok with this proposal.
But note that it will not necessarily prevent the INPUTS_SETTER_ONE_ITEM error from happening (I don't know if it is part of your goal). Depending on how the provided inputs are generated, the second input could be null for instance (and no other inputs are provided).

Reply:
Also fine for me